### PR TITLE
Assigment 4 Eduardo Álvarez

### PR DESCRIPTION
**Pull request by edualvareez**

### CODE REVIEW

**Comment by SerPablo** [comment]
Try 7.5! 
Hint 

?Subject   xxxxxxxxxxxxxxxxxxxxx. xxxxxxxx
 filter(?b != ?c)

File: Assignment4/e22c022- 22c022/task07.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+# TO DO
+
+ns = Namespace("http://somewhere#")
+from rdflib.plugins.sparql import prepareQuery
+
+#RDFLib
+
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+  print(s)
```
Comment (dgarijo) [inline Assignment4/e22c022- 22c022/task07.py:34]: not recursive

Comment (dgarijo) [reply Assignment4/e22c022- 22c022/task07.py:34]: You need to print ALL subclasses of living thing (recursively)

File: Assignment4/e22c022- 22c022/task07.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+#SPARQL
+
+query1 = """
+SELECT ?subclass
+WHERE {
+    ?subclass rdfs:subClassOf ns:LivingThing .
```
Comment (dgarijo) [inline Assignment4/e22c022- 22c022/task07.py:41]: same as above

File: Assignment4/e22c022- 22c022/task07.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+# TO DO
+
+#RDFLib
+
+for s,p,o in g.triples((None, RDF.type, ns.Person)):
+  print(s)
+for s1,p1,o1 in g.triples((None, RDFS.subClassOf, ns.Person)):
+  for s2,p2,o2 in g.triples((None, RDF.type, s1)):
+    print(s2)
```
Comment (dgarijo) [inline Assignment4/e22c022- 22c022/task07.py:62]: This only goes 1 level into subclasses. Hint: do a recursive function

File: Assignment4/e22c022- 22c022/task07.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+{
+    {
+      ?person rdf:type ns:Person.
+    }
+    UNION
+    {
+      ?subclass rdfs:subClassOf ns:Person.
+      ?person a ?subclass.
+    }
+}
```
Comment (dgarijo) [inline Assignment4/e22c022- 22c022/task07.py:78]: same as above. Use chained operators with `*`

File: Assignment4/e22c022- 22c022/task07.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+{
+    {
+        ?person rdf:type ns:LivingThing.
+        ?person ?property ?value.
+    }
+    UNION
+    {
+        ?subclass rdfs:subClassOf ns:LivingThing.
+        ?person a ?subclass.
+        ?person ?property ?value.
```
Comment (dgarijo) [inline Assignment4/e22c022- 22c022/task07.py:115]: you need to do the UNION with animal and Person, not living thing

File: Assignment4/e22c022- 22c022/task07.py [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    ?person ns:name ?personName.
+}
+'''
+
+# Visualizar los resultados
+for r in g.query(query4):
+    print(r)
+
+"""**Task 7.5: List the entities who know at least two other entities in the graph**"""
+
```
Comment (dgarijo) [inline Assignment4/e22c022- 22c022/task07.py:153]: missing!

**Review by dgarijo — CHANGES_REQUESTED** [review]

**Comment by anaigmo** [comment]
Entregado, corregido con algunos fallos, no se admiten más PR

**Pull request merged**